libs/tinygettext/src/dictionary_manager.cpp

```diff
@@ -258,41 +258,6 @@ DictionaryManager::set_filesystem(std::unique_ptr<FileSystem> filesystem_)
 {
   filesystem = std::move(filesystem_);
 }
-// ----------------------------------------------------------------------------
-/** This function converts a .po filename (e.g. zh_TW.po) into a language
- *  specification (zh_TW). On case insensitive file systems (think windows)
- *  the filename and therefore the country specification is lower case
- *  (zh_tw). It Converts the lower case characters of the country back to
- *  upper case, otherwise tinygettext does not identify the country
- *  correctly.
```

Comment:
this logic seems required no?

On windows the filenames could be `zh_hant_hk.po`

And thus won't be identified by tinygettext?

Reply:
You can see `Language::match` now compares language specs in lowercase.

For the purpose of `DictionaryManager::get_dictionary`, tinygettext compares pre-defined languages (they have correct cases) to languages from filenames, and then *uses a pre-defined language that is closest to the best file*. So we don’t really need to “correct” that so-called “country” spec, but rather match it loosely, ignoring case.

I’d test on Windows when I’m available. This fix works well on Linux (I have `zh_Han*.po` language files locally, that I’m working on)

Reply:
Ah yes, I see.